Error TS2307

**Describe the bug**
When building the project after installing tapi, get several TS2307 errors (e.g. `Error: node_modules/tapi.js/dist/es/index.d.ts:1:21 - error TS2307: Cannot find module '@/core/Builder' or its corresponding type declarations`).

May or may not be relevant, I got this error on an Angular 15 project.

**To Reproduce**
Install library using `npm i -D tapi.js`, then build/run the project (e.g. `ng serve` on Angular)

**Expected behavior**
Successful compilation

**Environment (please complete the following information):**
 - tapi version 0.5.0
 - node version 9.5.0
 - Angular version 15.2.0

**Typescript config file(s):**
```json
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "sourceMap": true,
    "declaration": false,
    "downlevelIteration": true,
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "ES2022",
    "module": "ES2022",
    "useDefineForClassFields": false,
    "lib": [
      "ES2022",
      "dom"
    ]
  },
  "angularCompilerOptions": {
    "enableI18nLegacyMessageIdFormat": false,
    "strictInjectionParameters": true,
    "strictInputAccessModifiers": true,
    "strictTemplates": true
  }
}
```

**Additional context**
I assume there is some (Babel?) syntax that is not supported universally in JavaScript. I could probably do a few searches to find how to compile with this syntax but put up this issue since other new users may encounter this issue.

--- a/package.json
+++ b/package.json
@@ -59,6 +59,7 @@
 		"eslint": "^8.29.0",
 		"jest": "^29.3.1",
 		"rollup": "^3.8.0",
+		"rollup-plugin-clear": "^2.0.7",
 		"rollup-plugin-dts": "^5.3.0",
 		"ts-jest": "^29.0.3",
 		"typescript": "^4.9.3"

--- a/package.json
+++ b/package.json
@@ -59,7 +59,7 @@
 		"eslint": "^8.29.0",
 		"jest": "^29.3.1",
 		"rollup": "^3.8.0",
-		"rollup-plugin-clear": "^2.0.7",
+		"rollup-plugin-delete": "^2.0.0",
 		"rollup-plugin-dts": "^5.3.0",
 		"ts-jest": "^29.0.3",
 		"typescript": "^4.9.3"

--- a/package.json
+++ b/package.json
@@ -23,7 +23,7 @@
 	"homepage": "https://tapi.js.org",
 	"main": "dist/cjs/index.js",
 	"module": "dist/es/index.js",
-	"types": "dist/es/index.d.ts",
+	"types": "dist/types.d.ts",
 	"scripts": {
 		"build": "rollup -c --silent",
 		"test": "jest",
@@ -59,6 +59,7 @@
 		"eslint": "^8.29.0",
 		"jest": "^29.3.1",
 		"rollup": "^3.8.0",
+		"rollup-plugin-dts": "^5.3.0",
 		"ts-jest": "^29.0.3",
 		"typescript": "^4.9.3"
 	},

--- a/rollup.config.cjs
+++ b/rollup.config.cjs
@@ -9,6 +9,8 @@ const del = require('rollup-plugin-delete')
 
 // Constants
 const pkg = require('./package.json')
+const tsConfigBase = require('./tsconfig.json')
+const SRC_DIR = tsConfigBase.compilerOptions.baseUrl
 const LIB_DIR = pkg.directories.lib
 
 function createBuildConfig(outSubDir, outFormat) {
@@ -16,10 +18,10 @@ function createBuildConfig(outSubDir, outFormat) {
 
 	return {
 		input: [
-			'src/index.ts',
-			'src/functions.ts',
-			'src/extensions.ts',
-			'src/decorators.ts',
+			SRC_DIR + '/index.ts',
+			SRC_DIR + '/functions.ts',
+			SRC_DIR + '/extensions.ts',
+			SRC_DIR + '/decorators.ts',
 		],
 		output:[
 			{

--- a/rollup.config.cjs
+++ b/rollup.config.cjs
@@ -5,12 +5,14 @@ const cjs = require('@rollup/plugin-commonjs')
 const json = require('@rollup/plugin-json')
 const resolve = require('@rollup/plugin-node-resolve')
 const alias = require('@rollup/plugin-alias')
+const clear = require('rollup-plugin-clear')
 
 // Constants
 const pkg = require('./package.json')
+const LIB_DIR = pkg.directories.lib
 
 function createBuildConfig(outSubDir, outFormat) {
-	const OUT_DIR = pkg.directories.lib + '/' + outSubDir
+	const OUT_DIR = LIB_DIR + '/' + outSubDir
 
 	return {
 		input: [
@@ -42,8 +44,8 @@ function createBuildConfig(outSubDir, outFormat) {
 			json(),
 			ts({
 				tsconfig: './tsconfig.build.json',
-				declarationDir: OUT_DIR + '/types',
 				outDir: OUT_DIR,
+				declarationDir: OUT_DIR + '/types',
 			}),
 		],
 	}
@@ -53,14 +55,22 @@ module.exports = [
 	createBuildConfig('cjs', 'cjs'),
 	createBuildConfig('es', 'es'),
 	{
-		input: 'dist/es/types/index.d.ts',
-		output: [{ file: 'dist/types.d.ts', format: 'es' }],
+		input: LIB_DIR + '/es/types/index.d.ts',
+		output: [{ file: LIB_DIR + '/types.d.ts', format: 'es' }],
 		plugins: [
 			dts.default({
 				tsconfig: './tsconfig.build.json',
 				compilerOptions: {
-					baseUrl: 'src'
+					baseUrl: 'src',
+					sourceMap: true,
+					declarationMap: true,
 				},
+			}),
+			clear({
+				targets: [
+					LIB_DIR + '/es/types',
+					LIB_DIR + '/cjs/types',
+				],
 			}),
 		],
 	},

--- a/rollup.config.cjs
+++ b/rollup.config.cjs
@@ -1,5 +1,6 @@
 // Plugins
 const ts = require('@rollup/plugin-typescript')
+const dts = require('rollup-plugin-dts')
 const cjs = require('@rollup/plugin-commonjs')
 const json = require('@rollup/plugin-json')
 const resolve = require('@rollup/plugin-node-resolve')
@@ -41,15 +42,26 @@ function createBuildConfig(outSubDir, outFormat) {
 			json(),
 			ts({
 				tsconfig: './tsconfig.build.json',
-				outDir: OUT_DIR,
 				declarationDir: OUT_DIR + '/types',
-				tsBuildInfoFile: OUT_DIR + '/' + '.tsbuildinfo'
+				outDir: OUT_DIR,
 			}),
 		],
 	}
 }
 
 module.exports = [
+	createBuildConfig('cjs', 'cjs'),
 	createBuildConfig('es', 'es'),
-	createBuildConfig('cjs', 'cjs')
+	{
+		input: 'dist/es/types/index.d.ts',
+		output: [{ file: 'dist/types.d.ts', format: 'es' }],
+		plugins: [
+			dts.default({
+				tsconfig: './tsconfig.build.json',
+				compilerOptions: {
+					baseUrl: 'src'
+				},
+			}),
+		],
+	},
 ]

--- a/rollup.config.cjs
+++ b/rollup.config.cjs
@@ -1,11 +1,11 @@
 // Plugins
 const ts = require('@rollup/plugin-typescript')
-const dts = require('rollup-plugin-dts')
+const dts = require('rollup-plugin-dts').default
 const cjs = require('@rollup/plugin-commonjs')
 const json = require('@rollup/plugin-json')
 const resolve = require('@rollup/plugin-node-resolve')
 const alias = require('@rollup/plugin-alias')
-const clear = require('rollup-plugin-clear')
+const del = require('rollup-plugin-delete')
 
 // Constants
 const pkg = require('./package.json')
@@ -58,7 +58,7 @@ module.exports = [
 		input: LIB_DIR + '/es/types/index.d.ts',
 		output: [{ file: LIB_DIR + '/types.d.ts', format: 'es' }],
 		plugins: [
-			dts.default({
+			dts({
 				tsconfig: './tsconfig.build.json',
 				compilerOptions: {
 					baseUrl: 'src',
@@ -66,11 +66,13 @@ module.exports = [
 					declarationMap: true,
 				},
 			}),
-			clear({
+			del({
 				targets: [
 					LIB_DIR + '/es/types',
 					LIB_DIR + '/cjs/types',
 				],
+				hook: 'buildEnd',
+				runOnce: true,
 			}),
 		],
 	},

--- a/rollup.config.cjs
+++ b/rollup.config.cjs
@@ -7,9 +7,10 @@ const alias = require('@rollup/plugin-alias')
 
 // Constants
 const pkg = require('./package.json')
-const OUT_DIR = pkg.directories.lib
 
-function createBuildConfig(outFormat, outSubDir) {
+function createBuildConfig(outSubDir, outFormat) {
+	const OUT_DIR = pkg.directories.lib + '/' + outSubDir
+
 	return {
 		input: [
 			'src/index.ts',
@@ -19,7 +20,7 @@ function createBuildConfig(outFormat, outSubDir) {
 		],
 		output:[
 			{
-				dir: OUT_DIR + '/' + outSubDir,
+				dir: OUT_DIR,
 				format: outFormat
 			},
 		],
@@ -40,8 +41,9 @@ function createBuildConfig(outFormat, outSubDir) {
 			json(),
 			ts({
 				tsconfig: './tsconfig.build.json',
-				outDir: OUT_DIR + '/' + outSubDir,
-				tsBuildInfoFile: OUT_DIR + '/' + outSubDir + '/' + '.tsbuildinfo'
+				outDir: OUT_DIR,
+				declarationDir: OUT_DIR + '/types',
+				tsBuildInfoFile: OUT_DIR + '/' + '.tsbuildinfo'
 			}),
 		],
 	}

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,5 +1,6 @@
 // Core
 import Builder from '@/core/Builder'
+import BuildConfiguration from '@/core/BuildConfiguration'
 import BuildableResource from "@/contracts/BuildableResource"
 
 // Decorators
@@ -7,6 +8,7 @@ import * as Properties from '@/decorators'
 
 export {
 	Builder,
+	BuildConfiguration,
 	BuildableResource,
 	Properties,
 }

--- a/tsconfig.build.json
+++ b/tsconfig.build.json
@@ -6,7 +6,6 @@
 		"declarationMap": true,
 		"sourceMap": true,
 		"rootDir": "src",
-		"composite": true,
 		"removeComments": true,
 		"skipLibCheck": false,
 	},

